### PR TITLE
Add a changelog

### LINKED ISSUE #47: Add more information to README about changes on this fork
I think it would help to have a bit more info on the README to explain what is different about this fork compared to the original repo. 

What do you think about having a condensed **changelog**, to show the various issues that have been fixed? This could help give other developers or users a clearer sense of what has changed, and give them confidence in what has _not_ changed (e.g. that the API remains the same).

It could either be a summary in the README, or perhaps a separate changelog. I'm inclined to go for the main README for visibility. I'd recommend the standard format from https://keepachangelog.com/en/1.0.0/

### PR DESCRIPTION
**Pull request by connortann**
Closes #47

### CODE REVIEW

File: CHANGELOG.md
Comment (thatlittleboy) [inline CHANGELOG.md:26]: IMO, this is more of a bug fix than a "change". Because I don't think the `sampling with replacement` behaviour was ever intended (it doesn't make sense in the context in which they were being used).

**Review by thatlittleboy — CHANGES_REQUESTED** [review]
Thanks for the initiative. Btw, let's agree that we will only update this file before release, by one of the maintainers. Rather than enforce it in every PR yea? It would become too messy otherwise

File: CHANGELOG.md
Comment (connortann) [reply CHANGELOG.md:21]: Thanks for the review comments, I'll push an update when I get some time.

> Thanks for the initiative. Btw, let's agree that we will only update this file before release, by one of the maintainers. Rather than enforce it in every PR yea? It would become too messy otherwise

I generally prefer advocate for adding in items to an "Unreleased" section on an ongoing basis, as recommended in [How can I reduce the effort required to maintain a changelog](https://keepachangelog.com/en/1.0.0/#effort). Benefits of an incremental approach are:
1. We don't accrue too much tech debt having to review diffs on each release, and
2. The changelog remains reasonably up-to-date at all times, without waiting for a release.

I might be missing something, what is the messiness that you allude to with this approach? The risk of annoying merge conflicts can be mitigated quite easily: we can add `CHANGELOG merge=union` to a `.gitattributes`, so that multiple additions are automatically concatenated together without raising a conflict.

Certainly, some PRs from new contributors might omit an entry to the changelog, so we will probably always need to do a bit of cleanup before making a release. I'd rather not have to gather it all from scratch each time though.

Comment (thatlittleboy) [reply CHANGELOG.md:21]: > we can add CHANGELOG merge=union to a .gitattributes, so that multiple additions are automatically concatenated together without raising a conflict.

Cool, I didn't know about this. The problem I had with my previous OSS project was that we would have a changelog.md, that in 1 PR looks like

```
- introduced xxx in #101
- introduced yyy in #102
- introduced zzz in #104
```

and in another that looks like 
```
- introduced xxx in #101
- introduced www in #103
- introduced zzz in #104
```
And merging PR 102 would then invalidate 103 because of a merge conflict.
Does setting the gitattributes resolve such problems?

If it does, then let's also add this change to this PR?

---

Agree with your pros. The above is really my only main concern.

I'll also add one last comment:

Based on what I've seen: probably most contributors won't add an entry to the CHANGELOG, with or without a contributing guideline / a reminder in the issue template. And I don't want to have to force this upon the contributors, 1) it adds one additional hurdle for them; and 2) IMO project admin should be handled by the maintainers, not the contributors. I'll rather they spend the time on writing good tests and making sure they pass. 3) There's also no guarantee that their changelog entry meets our changelog "style", e.g. grammatically correct, including the PR number.. so I'll much rather we control the changelog ourselves.

Comment (connortann) [reply CHANGELOG.md:21]: Sounds good! Let's put the onus on maintainers, rather than on each PR.

This topic is a bit prone to [bikeshedding](https://en.wiktionary.org/wiki/bikeshedding), which I'll try to avoid. The main thing is that we get it done :)

File: .gitattributes
Comment (thatlittleboy) [inline .gitattributes:1]: typo: consistent

**Review by thatlittleboy — CHANGES_REQUESTED** [review]

**Comment by connortann** [comment]
Good spot on typos, fixed now. FWIW, I'd be happy for other maintainers to just push more commits to this branch in PRs like this for obvious fixes like typos, formatting, etc.

**Pull request merged**
